src/renderer/components/dock/terminal/terminal.ts

```diff
@@ -63,15 +59,25 @@ export class Terminal {
     const { elem } = this;
 
     if (elem) {
-      Terminal.spawningPool.appendChild(elem);
+      this.dependencies.spawningPool.appendChild(elem);
```

Comment:
Food for thought: React portal could maybe do this in a better way (less brittle, more react, more unit-testability, etc.)?

Reply:
Maybe, I would have to read up on how react portal works. For the future.